docs/.vuepress/config.ts

```diff
@@ -105,3 +126,71 @@ export default defineUserConfig({
     },
   }),
 });
+
+function getAndBuildKitModuleTree(dir: string, child: string = "") {
+  const files = fs.readdirSync(path.join(dir, child), { withFileTypes: true })
+
+  const isKitModule = !!files.find(f => f.name === "README.md")
+      && child !== ""; // When no child is there, we are in the root kit folder, which contains a README too.
+
+  if (isKitModule) {
+    // If we found a kit module, we will copy its README over to the hub folder.
+    const source = path.join(dir, child, "README.md");
+    const destination = path.join(__dirname, "..", "hub", child);
+    const content = replaceReadMe(fs.readFileSync(source, "utf-8"), child);
+    fs.mkdirSync(destination, { recursive: true })
+    fs.writeFileSync(path.join(destination, "README.md"), content);
+
+    return [path.join("/", "hub", child)];
+  } else {
+    return files
+      .filter((x) => {
+        return x.isDirectory()
+            && treeSkip.indexOf(x.name) === -1
+            && !isKitModule // We do not allow nested kit modules
+      })
+      .map((x) => {
+        const nextChild = path.join(child, x.name);
+        return {
+          text: x.name,
+          collapsible: true,
+          children: [
+            ...getAndBuildKitModuleTree(dir, nextChild),
+          ],
+        };
+      });
+  }
+}
+
+/**
+ * This will inject a template with source code & installation to a README.
+ */
+function replaceReadMe(rawContent: string, module: string) {
+  // First we need to split up the content into frontmatter & non-frontmatter, as the frontmatter might contain comments
+  // which are the exact same syntax as Headers in markdown (# My YAML comment)
+  const frontMatterSplit = rawContent.split('---');
```

Comment:
I guess this works on the happy path... we should replace this with a proper markdown parser that splits the frontmatter :-) I'm pretty sure we've solved that problem somewhere else already. This is sth. we can fix in the future though